test/common/network/address_impl_test.cc

```diff
@@ -288,6 +289,20 @@ TEST(PipeInstanceTest, Basic) {
   EXPECT_EQ(nullptr, address.ip());
 }
 
+TEST(PipeInstanceTest, AbstractNamespace) {
+  std::unique_ptr<PipeInstance> address;
+#if defined(__linux__)
+  address = std::make_unique<PipeInstance>("@/foo");
+#else
+  EXPECT_THROW(address = std::make_unique<PipeInstance>("@/foo"), EnvoyException);
+  return;
```

Comment:
Instead of having an early return here, put all the EXPECTations below before the #else.